internal/registration/internal/domain/user_registration.go

```diff
@@ -0,0 +1,46 @@
+package domain
+
+import (
+	"errors"
+	"time"
+)
+
+var (
+	UserRegistrationCannotBeConfirmedMoreThanOnceError = errors.New("user can't be confirmed more than once")
+)
+
+func RegisterNewUser(firstName, lastName, password string, email UserRegistrationEmail) (*UserRegistration, error) {
+
+	return &UserRegistration{
+		ID:               NewUserRegistrationID(),
+		Status:           WaitForConfirmation,
+		Email:            email,
+		Name:             firstName + " " + lastName,
```

Comment:
What if I provide an empty first name and/or last name? Would those be considered correct names? We should at least check that firstName, lastName and password provided aren't empty.